x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/DocVectorTests.java

```diff
@@ -62,34 +64,53 @@ public void testNonDecreasingDescendingDocs() {
         docs.close();
     }
 
-    private static int MAX_BUILD_BREAKS_LIMIT = 1391;
+    private static long MAX_BUILD_BREAKS_LIMIT = -1;
+    private static long MAX_SHARD_SEGMENT_DOC_MAP_BREAKS = -1;
+
+    @Before
+    public void calculateBreakLimit() {
+        if (MAX_BUILD_BREAKS_LIMIT == -1) {
+            MAX_BUILD_BREAKS_LIMIT = BreakerTestUtil.findBreakerLimit(ByteSizeValue.ofMb(128), limit -> buildDocBlock(limit).close())
+                .getBytes();
+        }
+        if (MAX_SHARD_SEGMENT_DOC_MAP_BREAKS == -1) {
+            MAX_SHARD_SEGMENT_DOC_MAP_BREAKS = BreakerTestUtil.findBreakerLimit(ByteSizeValue.ofMb(128), limit -> {
+                try (DocBlock docBlock = buildDocBlock(limit)) {
+                    docBlock.asVector().shardSegmentDocMapForwards();
+                }
+            }).getBytes();
+        }
+    }
 
     public void testBuildBreaks() {
-        testBuildBreaks(ByteSizeValue.ofBytes(between(0, MAX_BUILD_BREAKS_LIMIT)));
+        testBuildBreaks(ByteSizeValue.ofBytes(randomLongBetween(0, MAX_BUILD_BREAKS_LIMIT)));
     }
 
     public void testBuildBreaksMax() {
         testBuildBreaks(ByteSizeValue.ofBytes(MAX_BUILD_BREAKS_LIMIT));
     }
```

Comment:
I don't think we need the MAX limit test if we're going to find this on the fly. The act of finding it confirms that it breaks.